ui-events-winit/src/pointer.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2025 the UI Events Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Support routines for converting pointer data from [`winit`].
+
+use ui_events::pointer::PointerButton;
+use winit::event::MouseButton;
+
+/// Try to make a [`PointerButton`] from a [`MouseButton`].
+///
+/// Because values of [`MouseButton::Other`] can start at 0, they are mapped
+/// to the arbitrary buttons B7..B32.
+/// Values greater than 25 will not be mapped.
+pub fn try_from_winit_button(b: MouseButton) -> Option<PointerButton> {
+    Some(match b {
+        MouseButton::Left => PointerButton::Primary,
+        MouseButton::Right => PointerButton::Secondary,
+        MouseButton::Middle => PointerButton::Auxiliary,
+        MouseButton::Back => PointerButton::X1,
+        MouseButton::Forward => PointerButton::X2,
+        MouseButton::Other(u) => match u {
+            0 => PointerButton::B7,
```

Comment:
Does this really start at 0 in `winit`? Or is it starting at 5 or 6?

Reply:
All I know is it's `u16`, I can look closer though. I think windows has defined buttons up to bit 6.

Reply:
The macOS code in `winit`:
```
    match unsafe { event.buttonNumber() } {
        0 => MouseButton::Left,
        1 => MouseButton::Right,
        2 => MouseButton::Middle,
        3 => MouseButton::Back,
        4 => MouseButton::Forward,
        n => MouseButton::Other(n as u16),
    }
```

Reply:
Wayland is:
```
fn wayland_button_to_winit(button: u32) -> MouseButton {
    // These values are coming from <linux/input-event-codes.h>.
    const BTN_LEFT: u32 = 0x110;
    const BTN_RIGHT: u32 = 0x111;
    const BTN_MIDDLE: u32 = 0x112;
    const BTN_SIDE: u32 = 0x113;
    const BTN_EXTRA: u32 = 0x114;
    const BTN_FORWARD: u32 = 0x115;
    const BTN_BACK: u32 = 0x116;

    match button {
        BTN_LEFT => MouseButton::Left,
        BTN_RIGHT => MouseButton::Right,
        BTN_MIDDLE => MouseButton::Middle,
        BTN_BACK | BTN_SIDE => MouseButton::Back,
        BTN_FORWARD | BTN_EXTRA => MouseButton::Forward,
        button => MouseButton::Other(button as u16),
    }
}
```
So we can't make any guarantees here ... hmm ...

Reply:
Seems that web sys is the odd one out, for defining the eraser as a button directly after X2

Reply:
The Wayland ones come from the linux headers and I think there are orderly ways to get the HID mouse button numbers from that.